modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/communicators/aah_async_comm.cc

```diff
@@ -70,192 +62,157 @@ AAH_ASynchronousCommunicator::Reset()
   data_initialized_ = false;
   upstream_data_initialized_ = false;
 
-  for (auto& message_flags : prelocI_message_received_)
-    message_flags.assign(message_flags.size(), false);
+  for (auto& rcv_flags : preloc_msg_received_)
+    rcv_flags.assign(rcv_flags.size(), false);
 
-  for (auto& message_flags : delayed_prelocI_message_received_)
-    message_flags.assign(message_flags.size(), false);
+  for (auto& rcv_flags : delayed_preloc_msg_received_)
+    rcv_flags.assign(rcv_flags.size(), false);
 }
 
 void
 AAH_ASynchronousCommunicator::BuildMessageStructure()
 {
   const auto& spds = fluds_.GetSPDS();
-  auto& fluds = dynamic_cast<AAH_FLUDS&>(fluds_);
-
-  // Predecessor locations
-  size_t num_dependencies = spds.GetLocationDependencies().size();
-
-  prelocI_message_count_.resize(num_dependencies, 0);
-  prelocI_message_size_.resize(num_dependencies);
-  prelocI_message_blockpos_.resize(num_dependencies);
-  prelocI_message_received_.clear();
+  const auto& fluds = dynamic_cast<AAH_FLUDS&>(fluds_);
 
-  for (int prelocI = 0; prelocI < num_dependencies; ++prelocI)
+  auto message_count_and_size = [this](const auto num_unknowns)
   {
-    size_t num_unknowns = fluds.GetPrelocIFaceDOFCount(prelocI) * num_groups_ * num_angles_;
     size_t message_count = num_angles_;
     if (num_unknowns * 8 > max_mpi_message_size_)
       message_count = ((num_unknowns * 8) + (max_mpi_message_size_ - 1)) / max_mpi_message_size_;
     size_t message_size = (num_unknowns + (message_count - 1)) / message_count;
+    return std::make_pair(message_count, message_size);
+  };
 
-    prelocI_message_count_[prelocI] = message_count;
-    prelocI_message_size_[prelocI].reserve(message_count);
-    prelocI_message_blockpos_[prelocI].reserve(message_count);
+  // Predecessor locations
+  const size_t num_dependencies = spds.GetLocationDependencies().size();
+  preloc_msg_data_.resize(num_dependencies);
+  preloc_msg_received_.resize(num_dependencies);
+
+  for (int i = 0; i < num_dependencies; ++i)
```

Comment:
`i` should be `size_t`, since `num_dependencies` is `size_t` as well.

Reply:
Hmm... I know I fixed that. Let me check and see if I forgot to push a commit.

Reply:
Strrange. Looks correct in the diff now and the comment shows as outdated.

Reply:
I see the outdated thing as well. github broken?

Reply:
I think I looked at the commit rather then the complete diff... It all looks good in the complete PR diff.